apps/dash-3d-image-partitioning/app.py

```diff
@@ -0,0 +1,935 @@
+import dash
+from dash.dependencies import Input, Output, State, ClientsideFunction
+import dash_html_components as html
+import dash_core_components as dcc
+import plotly.graph_objects as go
+from skimage import data, img_as_ubyte, segmentation, measure
+from dash_canvas.utils import array_to_data_url
+import plotly.graph_objects as go
+import plot_common
+import image_utils
+import numpy as np
+from nilearn import image
+import nibabel as nib
+import plotly.express as px
+import shape_utils
+from app_utils import get_env
+from sys import exit
+import io
+import base64
+import skimage
+import time
+
+DEBUG_MASK = False
+DEFAULT_STROKE_COLOR = px.colors.qualitative.Light24[0]
+DEFAULT_STROKE_WIDTH = 5
+# the scales for the top and side images (they might be different)
+# TODO: If the width and height scales are different, strange things happen? For
+# example I have observed the masks getting scaled unevenly, maybe because the
+# axes are actually scaled evenly (fixed to the x axis?) but then the mask gets
+# scaled differently?
+hwscales = [(2, 2), (2, 2)]
+# the number of dimensions displayed
+NUM_DIMS_DISPLAYED = 2  # top and side
+# the color of the triangles displaying the slice number
+INDICATOR_COLOR = "DarkOrange"
+DISPLAY_BG_COLOR = "darkgrey"
+
+# A string, if length non-zero, saves superpixels to this file and then exits
+SAVE_SUPERPIXEL = get_env("SAVE_SUPERPIXEL", default="")
+# A string, if length non-zero, loads superpixels from this file
+LOAD_SUPERPIXEL = get_env("LOAD_SUPERPIXEL", default="")
+
+
+def make_seg_image(img):
+    """ Segment the image, then find the boundaries, then return an array that
+    is clear (alpha=0) where there are no boundaries. """
+    segb = np.zeros_like(img).astype("uint8")
+    seg = segmentation.slic(
+        img, start_label=1, multichannel=False, compactness=0.1, n_segments=300
+    )
+    segb = segmentation.find_boundaries(seg).astype("uint8")
+    segl = image_utils.label_to_colors(
+        segb, colormap=["#000000", "#E48F72"], alpha=[0, 128], color_class_offset=0
+    )
+    return (segl, seg)
+
+
+def make_default_figure(
+    images=[],
+    stroke_color=DEFAULT_STROKE_COLOR,
+    stroke_width=DEFAULT_STROKE_WIDTH,
+    shapes=[],
```

Comment:
the `shapes` parameter is not used in this app, could you please remove it and the corresponding code?

Reply:
:+1: